crates/superposition/src/auth.rs

```diff
@@ -0,0 +1,150 @@
+mod authenticator;
+mod oidc;
+
+use std::{
+    future::{ready, Ready},
+    sync::Arc,
+};
+
+use actix_web::{
+    body::{BoxBody, EitherBody},
+    dev::{forward_ready, Service, ServiceRequest, ServiceResponse, Transform},
+    http::header,
+    web::Data,
+    Error, HttpMessage, Scope,
+};
+use aws_sdk_kms::Client;
+use futures_util::future::LocalBoxFuture;
+use service_utils::{
+    db::utils::get_oidc_client_secret,
+    helpers::get_from_env_unsafe,
+    service::types::{AppEnv, AppState},
+};
+use superposition_types::User;
+use url::Url;
+
+use self::authenticator::Authenticator;
+
+pub struct AuthMiddleware<S> {
+    service: S,
+    auth_handler: AuthHandler,
+}
+
+impl<S, B> Service<ServiceRequest> for AuthMiddleware<S>
```

Comment:
nab:
```suggestion
impl<S, B> Service<ServiceRequest> for OIDCAuthMiddleware<S>
```

Reply:
This is a generic middleware, the type/protocol is not going to just be OIDC.